suite-native/atoms/src/Sheet/BottomSheet.tsx

```diff
@@ -1,93 +1,87 @@
+import { BottomSheetHeader } from './BottomSheetHeader';
 
 type BottomSheetProps = {
     isVisible: boolean;
     onVisibilityChange: (isVisible: boolean) => void;
     children: ReactNode;
     title: string;
-    onBackArrowClick?: () => void;
 };
 
 type WrapperStyleProps = {
     insetBottom: number;
 };
-
 const sheetWrapperStyle = prepareNativeStyle<WrapperStyleProps>((utils, { insetBottom }) => ({
     backgroundColor: utils.colors.gray100,
     borderTopLeftRadius: utils.borders.radii.large,
     borderTopRightRadius: utils.borders.radii.large,
     paddingBottom: Math.max(insetBottom, utils.spacings.medium),
-}));
-
-const CLOSE_BUTTON_SIZE = 40;
-
-const closeButtonStyle = prepareNativeStyle(utils => ({
-    backgroundColor: utils.colors.gray100,
-    borderRadius: utils.borders.radii.round,
-    height: CLOSE_BUTTON_SIZE,
-    width: CLOSE_BUTTON_SIZE,
-    justifyContent: 'center',
-    alignItems: 'center',
-}));
-
-const sheetHeaderStyle = prepareNativeStyle(utils => ({
-    flexDirection: 'row',
-    justifyContent: 'space-between',
-    alignItems: 'center',
-    paddingHorizontal: utils.spacings.medium,
-    paddingVertical: utils.spacings.medium,
+    height: 600,
 }));
 
 const sheetWithOverlayStyle = prepareNativeStyle(_ => ({
     flex: 1,
     justifyContent: 'flex-end',
 }));
 
-export const BottomSheet = ({
-    isVisible,
-    onVisibilityChange,
-    title,
-    onBackArrowClick,
-    children,
-}: BottomSheetProps) => {
-    const { applyStyle } = useNativeStyles();
-    const insets = useSafeAreaInsets();
-    const {
-        animatedSheetWithOverlayStyle,
-        animatedSheetWrapperStyle,
-        closeSheetAnimated,
-        resetSheetAnimated,
-        panGestureEvent,
-    } = useBottomSheetAnimation({ onVisibilityChange, isVisible });
+export const BottomSheet = forwardRef<any, BottomSheetProps>(
+    ({ isVisible, onVisibilityChange, title, children }, ref) => {
+        const { applyStyle } = useNativeStyles();
+        const insets = useSafeAreaInsets();
+        const [isCloseScrollEnabled, setIsCloseScrollEnabled] = useState(true);
+        const {
+            animatedSheetWithOverlayStyle,
+            animatedSheetWrapperStyle,
+            closeSheetAnimated,
+            resetSheetAnimated,
+            panGestureEvent,
+            scrollEvent,
+        } = useBottomSheetAnimation({
+            onVisibilityChange,
+            isVisible,
+            isCloseScrollEnabled,
+            onIsCloseScrollEnabled: (isCloseScrollEnabled: boolean) => {
+                setIsCloseScrollEnabled(isCloseScrollEnabled);
+            },
+        });
+        const panGestureRef = useRef();
+        const scrollViewRef = useRef();
 
-    useEffect(() => {
-        if (isVisible) {
-            resetSheetAnimated();
-        }
-    }, [isVisible, resetSheetAnimated]);
+        useImperativeHandle(ref, () => ({
+            close: () => closeSheetAnimated(),
```

Comment:
Why do we need this? This imperative stuff makes sense only in some specific cases (like input stuff) and when you need things extra versatile which we don't need here. We can easily handle this much cleaner and easier with just `isVisible` prop.

This will also actually introduce bugs, for example what will happen when someone will call `close` but keep `isVisible=true`? Same problem will be with potentional `open` handler.

You can check https://reactnative.dev/docs/modal#visible RN Modal component and you can see they are also avoiding this imperative stuff in very similar component.

Reply:
Both seems like overkill. I don't understand why animation can't be triggered by `isVisible` prop? You can simply use something like this
```tsx
useEffect(() => {
  if(isVisible) {
    openSheetAnimated()
  } else {
    closeSheetAnimated()
  }
}, [isVisible])
```

Reply:
Because by the time this prop is distributed to the BottomSheet component, the modal is already closed. We could have a call and I will explain it further.